src/EnergyEfficientAgent.cpp

```diff
@@ -228,7 +231,8 @@ namespace geopm
             if (m_last_region_info[ctl_idx].hash != current_region_info.hash ||
                 m_last_region_info[ctl_idx].count != current_region_info.count) {
                 if (current_region_info.hash != GEOPM_REGION_HASH_UNMARKED &&
-                    current_region_info.hint != GEOPM_REGION_HINT_NETWORK) {
+                    current_region_info.hint != GEOPM_REGION_HINT_NETWORK &&
```

Comment:
might be nice to have a private method:

```
EnergyEfficientAgent::do_learning(uint64_t hash, uint64_t hint)
{
    return (hash != GEOPM_REGION_UNMARKED &&
            hint != GEOPM_REGION_HINT_NETWORK &&
            hint != GEOPM_REGION_HINT_IGNORE);
}
```
That way this logic doesn't get repeated twice.

Reply:
done